packages/firebase_database/lib/src/event.dart

```diff
@@ -34,3 +34,33 @@ class DataSnapshot {
   /// Returns the contents of this data snapshot as native types.
   dynamic get value => _data['value'];
 }
+
+class MutableData {
+  final Map<String, dynamic> _data;
+  MutableData._(this._data);
```

Comment:
Would it be better to clone the incoming map like:

    _data = new Map<String, dynamic>.from(data);

The map has no setter so it can't be changed and we can comment that we don't intend for it to be final?

Reply:
I was going to suggest making a copy, but I think it's actually even more confusing this way. `Map.from` looks like a deep clone but actually it only performs a shallow copy, so you'd have a `MutableData` where you can set properties of the `value` without having side effects, but mutating existing values in the map can cause your changes to propagate to other `MutableData` elsewhere. I think it's better to go all-or-nothing and warn callers that `MutableData` reserves the right to mutate its source data.

I think it's fine as-is with a comment added.